src/manifold/src/quickhull.cpp

```diff
@@ -821,7 +821,7 @@ bool QuickHull::addPointToFace(typename MeshBuilder::Face& f,
 
 // Wrapper to call the QuickHull algorithm with the given vertex data to build
 // the Impl
-void Manifold::Impl::Hull(const std::vector<vec3>& vertPos) {
+void Manifold::Impl::Hull(const Vec<vec3>& vertPos) {
```

Comment:
I will have a look at this when I get back to HK. This sounds really weird.

Reply:
OK I checked, this is due to some implicit type conversion and lifetime stuff. C++ automatically use the `Vec(const std::vector<T> &vec)` constructor to convert the `std::vector<vec3>& vertPos` into a `Vec<vec3>`. However, the problem is that immediately after the conversion it is being dropped, so what we get is just a dangling pointer as `VecView`. I don't think there is a good way to avoid this? This is why rust exists...

Reply:
lol, ugh. Sure there aren't at least some best practices with `Vec` to avoid this kind of thing? Is there really no constructors we can `delete` to keep it from compiling like that?

Reply:
It seems that there is no way to avoid the automatic temporary object creation when putting something convertible to `T` into a parameter that expects `const T&`. We can require conversions to `T` to be explicit, but not sure if that is good enough to avoid this kind of footgun.